core/services/beacon/settings.py

```diff
@@ -152,3 +152,28 @@ def migrate(self, data: Dict[str, Any]) -> None:
             logger.error(f"unable to update SettingsV1 to SettingsV2: {e}")
 
         data["VERSION"] = SettingsV2.VERSION
+
+
+class SettingsV3(SettingsV2):
+    VERSION = 3
+
+    def __init__(self, *args: str, **kwargs: int) -> None:
+        super().__init__(*args, **kwargs)
+        self.VERSION = SettingsV3.VERSION
+
+    def migrate(self, data: Dict[str, Any]) -> None:
+        if data["VERSION"] == SettingsV3.VERSION:
+            return
+
+        if data["VERSION"] < SettingsV3.VERSION:
+            super().migrate(data)
```

Comment:
Since the base class needs to be V1, we run the migration from V2.
```suggestion
            SettingsV2().migrate(data)
```